TestModels/SimpleTypes/SimpleBlob/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs

```diff
@@ -0,0 +1,546 @@
+#![allow(warnings, unconditional_panic)]
+#![allow(nonstandard_style)]
+pub use dafny_runtime;
+pub use dafny_standard_library;
+pub use dafny_standard_library::implementation_from_dafny::*;
+
+pub mod r#_simple_dtypes_dsmithyblob_dinternaldafny_dtypes {
+    /* datatype DafnyCallEvent<I, O> = DafnyCallEvent(input: I, output: O) */
+    #[derive(Clone)]
+    pub struct DafnyCallEvent<I: ::dafny_runtime::DafnyType, O: ::dafny_runtime::DafnyType> {
+        input: I,
+        output: O,
+    }
+    impl<I: ::dafny_runtime::DafnyType, O: ::dafny_runtime::DafnyType> ::core::fmt::Debug
+        for DafnyCallEvent<I, O>
+    {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            f.debug_struct("DafnyCallEvent")
+                .field("input", &self.input)
+                .field("output", &self.output)
+                .finish()
+        }
+    }
+    impl<I: ::dafny_runtime::DafnyType, O: ::dafny_runtime::DafnyType> ::dafny_runtime::DafnyPrint
+        for DafnyCallEvent<I, O>
+    {
+        fn fmt_print(&self, f: &mut std::fmt::Formatter<'_>, in_seq: bool) -> std::fmt::Result {
+            write!(
+                f,
+                "simple.types.smithyblob.internaldafny.types.DafnyCallEvent("
+            )?;
+            self.input.fmt_print(f, false)?;
+            write!(f, ",")?;
+            self.output.fmt_print(f, false)?;
+            write!(f, ")")
+        }
+    }
+    impl<I: ::dafny_runtime::DafnyTypeEq, O: ::dafny_runtime::DafnyTypeEq>
+        PartialEq<DafnyCallEvent<I, O>> for DafnyCallEvent<I, O>
+    {
+        fn eq(&self, other: &DafnyCallEvent<I, O>) -> bool {
+            self.input == other.input && self.output == other.output
+        }
+    }
+    impl<I: ::dafny_runtime::DafnyTypeEq, O: ::dafny_runtime::DafnyTypeEq> Eq for DafnyCallEvent<I, O> {}
+    impl<I: ::dafny_runtime::DafnyTypeEq, O: ::dafny_runtime::DafnyTypeEq> ::core::hash::Hash
+        for DafnyCallEvent<I, O>
+    {
+        fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+            self.input.hash(state);
+            self.output.hash(state);
+        }
+    }
+
+    /*
+     datatype GetBlobInput = | GetBlobInput (
+      nameonly value: Option<blob> := Option.None
+    )
+    */
+    #[derive(Clone)]
+    pub enum GetBlobInput {
+        GetBlobInput {
+            value: ::std::rc::Rc<
+                super::r#_Wrappers_Compile::Option<::std::vec::Vec<::std::primitive::u8>>,
```

Comment:
Since the Dafny type here is Option<seq<uint8>>, the generated type will be
```suggestion
                super::r#_Wrappers_Compile::Option<::dafny_runtime::Sequence<::std::primitive::u8>>,
```